components/Buttons/src/MDCButton.m

```diff
@@ -550,6 +578,56 @@ - (void)setElevation:(CGFloat)elevation forState:(UIControlState)state {
   }
 }
 
+#pragma mark - Border Color
+
+- (UIColor *)borderColorForState:(UIControlState)state {
+  return _borderColors[@(state)];
+}
+
+- (void)setBorderColor:(UIColor *)borderColor forState:(UIControlState)state {
+  if (!_borderColors) {
+    _borderColors = [NSMutableDictionary dictionary];
+  }
+  _borderColors[@(state)] = borderColor;
+
+  [self updateBorderColor];
+}
+
+- (void)updateBorderColor {
+  UIColor *color = _borderColors[@(self.state)];
+  if (!color && self.state != UIControlStateNormal) {
```

Comment:
Consider adding a comment here and in updateBorderWidth
// Use value for UIControllerStateNormal if self.state isn't defined